src/main/java/io/temporal/internal/sync/WorkflowRetryerInternal.java

```diff
@@ -186,7 +186,6 @@ public static void retry(
   }
 
   private static RetryOptions getRetryOptionsSideEffect(String retryId, RetryOptions options) {
-    options = RetryOptions.newBuilder(options).validateBuildWithDefaults();
```

Comment:
I'm confused why no validation here?  I thought this is also all client-side.

Reply:
This records retry options as they are, to ensure determinism if they change.